src/store/appReducer.js

```diff
@@ -0,0 +1,84 @@
+import { getUKFormattedDate } from '../utils/helpers';
+
+const appReducer = (state, action) => {
+  switch (action.type) {
+    case 'RESTORE_EXPENSES':
+      return { ...state, expenses: action.payload };
+
+    case 'ADD_EXPENSE_SUCCESS':
+      const expense = action.payload;
+      const expenseMonth = getUKFormattedDate(expense.date, { year: 'numeric', month: 'long' });
+
+      return {
+        ...state,
+        lastAddedExpense: expense,
+        isConfirmMonthModalOpen: expenseMonth !== state.selectedMonth,
+        didErrorOccur: false,
+        isAlertOpen: true,
+        userAction: 'add_expense',
+      };
+
+    case 'ADD_EXPENSE_FAIL':
+      return {
+        ...state,
+        didErrorOccur: true,
+        isAlertOpen: true,
+        userAction: 'add_expense',
+      };
+
+    case 'EDIT_EXPENSE':
```

Comment:
if there are `SUCCESS` and `FAIL` versions of `ADD_EXPENSE`, why not for `EDIT_EXPENSE`?

Reply:
The `SUCCESS` and `FAIL` variants of `DELETE_EXPENSE` and `EDIT_EXPENSE` are essentially the same in terms of which state variables they modify and what values are set (only `didErrorOccur` is different). Having two variants felt redundant.

`ADD_EXPENSE` on the other hand makes significantly different changes to state depending on whether the add operation succeeds or fails.

Reply:
yep. understand maybe my comment becomes obsolete if we decide to tackle `userAction`